code/game/objects/items/devices/flash.dm

```diff
@@ -64,6 +64,16 @@
 	var/flashfail = 0
 
 	if(iscarbon(M))
+		var/dionatype = M.is_diona()
+		if (dionatype)

```

Comment:
Small optimization here.
```
if (M.is_diona())
```
is more efficient, as you're not using `dionatype` anywhere else.